### PR TITLE
Make torch_mlir compatible with binary PyTorch installations.

### LINKED ISSUE #138: Problems with Torch graph bindings
Running into an issue

```
silvasean@silvasean0:~/pg/mlir-npcomp/mlir-npcomp$ source .env; python frontends/pytorch/test/graph_export/test_script_add3.py
Traceback (most recent call last):
  File "frontends/pytorch/test/graph_export/test_script_add3.py", line 21, in <module>
    def add3(t0, t1, t2):
TypeError: import_function(): incompatible function arguments. The following argument types are supported:
    1. (self: _torch_mlir.ModuleBuilder, arg0: torch::jit::StrongFunctionPtr) -> torch::jit::StrongFunctionPtr

Invoked with: <_torch_mlir.ModuleBuilder object at 0x7f81c3e8d0b0>, <torch.jit.ScriptFunction object at 0x7f81a8cb36d0>
```

1. For some reason, on my system pybind11.h is in two places, and despite our efforts in the CMakeLists.txt file the non-torch one seems to get used for the torch bindings

```
/usr/local/google/home/silvasean/.local/lib/python3.8/site-packages/pybind11/include/pybind11/pybind11.h
/usr/local/google/home/silvasean/.local/lib/python3.8/site-packages/torch/include/pybind11/pybind11.h
```

2. Even after I fix that by manually commenting out the necessary CMakeLists.txt lines, I get the same issue. Unclear why.

### PR DESCRIPTION
**Pull request by stellaraccident**
* This has been anticipated for a long time in that it is quite hard to keep C++ binary compatibility across a system landscape as diverse as PyTorch, LLVM, and this project. This is why we based the PyTorch extension on the MLIR and NPCOMP C APIs only: that is the only sane linkage story for the entire matrix.
* Removes the few LLVM'isms in torch_mlir that had snuck in, using either STL or PyTorch support utilities. The new rule here is that LLVM C++ includes are forbidden at this level and (as stated in the design), torch_mlir should use the PyTorch runtime and support libraries (not introduce an incidental C++ dependency on LLVM).
* Also deletes mnist-playground as it was proving impossible to keep the grid of PyTorch vs system ABI divisions functioning. I am open to a less drastic course here (optional/disabled by default?)
* This gets us pretty close to just using PyTorch's extension builder API, which will be nice for distribution (i.e. it integrates well with the PyTorch ecosystem for deployment). I ended up just simplifying the in-tree CMake support for now.
* Fixes #138

### CODE REVIEW

File: cmake/modules/ConfigurePyTorch.cmake [hunk 1]
```diff
@@ -1,11 +1,35 @@
-function(ProbeForPyTorchInstall)
+# NpcompFindPyTorch
+# Calls find_package on Torch and does any needed post-processing.
+# The master_enabled flag can be OFF, ON or OPTIONAL.
+macro(NpcompFindPyTorch master_enable)
```
Comment (silvasean) [inline cmake/modules/ConfigurePyTorch.cmake:4]: not sure what "master" means in this context. Maybe "enable_setting"?

File: cmake/modules/ConfigurePyTorch.cmake [hunk 2]
```diff
@@ -15,8 +39,42 @@ function(ProbeForPyTorchInstall)
...
-    # PyTorch stashes its installed .cmake files under share/cmake/Torch.
-    set(Torch_ROOT "${PYTORCH_PACKAGE_DIR}/share/cmake/Torch"
-        CACHE STRING "Torch package root")
+    set(Torch_ROOT "${PYTORCH_PACKAGE_DIR}" CACHE STRING
+        "Torch configure directory" FORCE)
+  endif()
+endfunction()
+
+# NpcompConfigurePyTorch
+# Performs post-find configuration of PyTorch flags. Most of this comes down
```
Comment (silvasean) [inline cmake/modules/ConfigurePyTorch.cmake:48]: post-find is a little terse. "Performs configuration of C++ compiler flags for the npcomp source files after PyTorch has been found"?

File: frontends/pytorch/csrc/CMakeLists.txt [hunk 3]
```diff
@@ -1,51 +1,49 @@
...
+  builder/graph_importer.cpp
+  builder/module_builder.cpp
+  builder/python_bindings.cpp
   init_python_bindings.cpp
   )
+
 target_link_libraries(NPCOMPTorchMLIRExt
+  # NPCOMP shared library.
+  # TODO: Debug why order matters here (if NPCOMP is included last a large
+  # amount of LLVM/MLIR/NPCOMP ends up compiled into this library).
```
Comment (silvasean) [inline frontends/pytorch/csrc/CMakeLists.txt:26]: "into this library" or "into executables linking this library"?

File: frontends/pytorch/csrc/builder/module_builder.cpp [hunk 4]
```diff
@@ -100,9 +99,9 @@ ModuleBuilder::importFunction(torch::jit::StrongFunctionPtr function) {
   auto inserter = createInserter();
   GraphImporter::MlirMappingOptions mappingOptions{
       context,
-      llvm::None, // genericFuncName (default to auto)
-      llvm::None, // funcName (default to auto)
-      typeMapper, inserter,
+      c10::nullopt, // genericFuncName (default to auto)
+      c10::nullopt, // funcName (default to auto)
+      typeMapper,   inserter,
```
Comment (silvasean) [inline frontends/pytorch/csrc/builder/module_builder.cpp:104]: nit: weird extra space

File: tools/mnist-playground/README.md [hunk 5]
```diff
@@ -1,42 +0,0 @@
-# mnist-playground
-
-This is intended to be a short-lived "playground" for doing various experiments, guided by a real model use case, for improving the npcomp reference backend.
-
-It's expected that utilities developed here will graduate to a more general utility or that this utility will be obsoleted by Python-driven flows once those come online.
```
Comment (silvasean) [inline tools/mnist-playground/README.md:5]: yay for python-driven flows coming online and this code being obsoleted!!!

**Review by silvasean — APPROVED** [review]

File: frontends/pytorch/csrc/builder/module_builder.cpp (hunk 4 above)
Comment (stellaraccident) [reply frontends/pytorch/csrc/builder/module_builder.cpp:104]: Weird. Removed the trailing comma and clang-format had a better time.

**Pull request merged**